engine/commands/model_upd_cmd.cc

```diff
@@ -0,0 +1,127 @@
+#include "model_upd_cmd.h"
+
+#include "utils/logging_utils.h"
+
+namespace commands {
+
+ModelUpdCmd::ModelUpdCmd(std::string model_handle)
+    : model_handle_(std::move(model_handle)) {}
+
+void ModelUpdCmd::Exec(
+    const std::unordered_map<std::string, std::string>& options) {
+  try {
+    auto model_entry = model_list_utils_.GetModelInfo(model_handle_);
+    yaml_handler_.ModelConfigFromFile(model_entry.path_to_model_yaml);
+    model_config_ = yaml_handler_.GetModelConfig();
+
+    for (const auto& [key, value] : options) {
+      if (!value.empty()) {
+        UpdateConfig(key, value);
+      }
+    }
+
+    yaml_handler_.UpdateModelConfig(model_config_);
+    yaml_handler_.WriteYamlFile(model_entry.path_to_model_yaml);
+    CLI_LOG("Successfully updated model ID '" + model_handle_ + "'!");
+  } catch (const std::exception& e) {
+    CLI_LOG("Failed to update model with model ID '" + model_handle_ +
+            "': " + e.what());
+  }
+}
+
+void ModelUpdCmd::UpdateConfig(const std::string& key,
+                               const std::string& value) {
+  static const std::unordered_map<
```

Comment:
Hmm, I think there's a better way. We can define a function: UpdateModelConfig(const std::string& attr, std::variant<std::string, explicit_int, explicit_bool>) and handle the update logic there.

However if this works for you then I'm fine with it.

Reply:
I think should keep it this way. We don't know which format user will input, so need to handle it for specific cases